type.go

```diff
@@ -469,7 +469,7 @@ func CompareTypes(a, b Type) int {
 		}
 		return 0
 	case ErrorKind:
-		ea, eb := a.(*TypeError), b.(*TypeError)
+		ea, eb := TypeUnder(a).(*TypeError), TypeUnder(b).(*TypeError)
```

Comment:
This had to be fixed because it was causing a parquet boomerang value when creating a union type of a named error.

Reply:
The `MapKind`, `UnionKind`, and `EnumKind` cases all need the same treatment. But rather than add `TypeUnder` calls in each, how about adding `a, b = TypeUnder(a), TypeUnder(b)` before the `switch` statement?